mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessage.java

```diff
@@ -39,17 +39,17 @@
 public class DeletedMessage {
     public static class Builder {
         @FunctionalInterface
-        interface RequireMessageId<T> {
+        public interface RequireMessageId<T> {
```

Comment:
Your history is dirty: how does this relate to `JAMES-2663 Adding action parameter to restoring API path ` ?